core/src/lib.rs

```diff
@@ -173,6 +175,55 @@ impl Default for Attribute {
     }
 }
 
+/// Map `String` idents (`:db/ident`) to positive integer entids (`1`).
+pub type IdentMap = BTreeMap<String, Entid>;
+
+/// Map positive integer entids (`1`) to `String` idents (`:db/ident`).
+pub type EntidMap = BTreeMap<Entid, String>;
+
+/// Map attribute entids to `Attribute` instances.
+pub type SchemaMap = BTreeMap<Entid, Attribute>;
+
+/// Represents a Mentat schema.
+///
+/// Maintains the mapping between string idents and positive integer entids; and exposes the schema
+/// flags associated to a given entid (equivalently, ident).
+///
+/// TODO: consider a single bi-directional map instead of separate ident->entid and entid->ident
+/// maps.
+#[derive(Clone, Debug, Default, Eq, Hash, Ord, PartialOrd, PartialEq)]
+pub struct Schema {
+    /// Map entid->ident.
+    ///
+    /// Invariant: is the inverse map of `ident_map`.
+    pub entid_map: EntidMap,
+
+    /// Map ident->entid.
+    ///
+    /// Invariant: is the inverse map of `entid_map`.
+    pub ident_map: IdentMap,
+
+    /// Map entid->attribute flags.
+    ///
+    /// Invariant: key-set is the same as the key-set of `entid_map` (equivalently, the value-set of
+    /// `ident_map`).
+    pub schema_map: SchemaMap,
+}
+
+impl Schema {
+    pub fn get_ident(&self, x: &Entid) -> Option<&String> {
+        self.entid_map.get(x)
+    }
+
+    pub fn get_entid(&self, x: &String) -> Option<&Entid> {
+        self.ident_map.get(x)
+    }
+
+    pub fn attribute_for_entid(&self, x: &Entid) -> Option<&Attribute> {
```

Comment:
@ncalexan: for the record, I didn't move the `require_` methods because that would require schlepping all of the `db` error machinery into `core`. The error kinds include rusqlite errors like `BadSQLValuePair`, and I don't know how `error_chain!` would handle merging two sets of errors from two different crates. So… I left it, because I don't need 'em!